### ISSUE TITLE
False positive of `no-empty-collection`

### ISSUE THREAD

**Issue by zloirock** [SonarSource/eslint-plugin-sonarjs#260]
```js
  const set = new Set();

  function filter(method, list) {
    for (const ns of list) {
      for (const name of modulesList) {
        if (name === ns || name.startsWith(`${ ns }.`)) {
          set[method](name); // Review this usage of "set" as it can only be empty here
        }
      }
    }
  }

  filter('add', modules);
  filter('delete', exclude);

  modules = modulesList.filter(it => set.has(it)); // Review this usage of "set" as it can only be empty here
```

Seems cases of calling methods when the method can't be clearly recognized as access or iteration should be ignored and the next checks should be discarded.

**Comment by vilchik-elena** [comment]
I don't see where `set` gets populated with values, could you clarify?

**Comment by zloirock** [comment]
> `filter('add', modules)`

**Comment by vilchik-elena** [comment]
Ok, I see. Tbh at first sight it seems to be a corner case, pretty rare and not trivial to fix. I will keep ticket open until next sprint, may be by that time we understand if it is worth fixing.

**Comment by Wohops** [comment]
This issue has been migrated to Jira. [ESLINTJS-13](https://sonarsource.atlassian.net/browse/ESLINTJS-13)
